### DISCUSSION TITLE
Gradle version catalog (libs.versions.toml) git conflicts

### DISCUSSION THREAD

**Discussion by jaroslavhavelik** [renovatebot/renovate#26251] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.83.3-slim

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### What is your question?

We are using Renovate for multiple projects with version catalogs ([libs.versions.toml](https://developer.android.com/build/migrate-to-catalogs)). We have scheduled a bitbucket pipeline that checks the versions and creates PRs for each dependency.
The issue is that versions are located in a single file and when two versions are on subsequent lines merging one of the PRs results in conflict in the second one.

What is the expected workflow after this happens (I'm sure that a similar thing can happen in package.json dependencies)? 

As we are self-hosting renovate I haven't found how to do automatic rebase and rebasing updates one by one seems like a big overhead when the result is just changing one version number.

**Comment by rarkins** [comment]
Renovate will automatically rebase any conflicted dependency. To reduce how often this happens, you can do things like:
- limit concurrency of updates
- group together updates

**Reply by jaroslavhavelik** [reply]
How exactly I will achieve a rebase of conflicted dependencies? As far as I understand "self-hosting" Renovate on Bitbucket Cloud essentially means having a pipeline that runs in some kind of period. It's not a running instance that can check if some PR was conflicted.
Do you have some examples how to self-host Renovate on the Bitbucket cloud? I haven't found many information except:
- https://docs.renovatebot.com/modules/platform/bitbucket/
- https://dev.to/pgburrack/bitbucket-dependency-bot-using-renovate-5d3b

Can you advise?

**Reply by rarkins** [reply, marked as answer]
Renovate needs to run in order to deconflict any conflicted PRs. So for example if you run it hourly, then you'd expect to wait up to one hour for deconflicting.

There's nothing special or specific about hosting for Bitbucket Cloud other than the basic variables in the docs.

We (Mend.io) plan to offer a free hosted version of Renovate for Bitbucket Cloud in early 2024, which will run similarly to the current app for github.com, including listening to webhooks and running based on events.

**Reply by jaroslavhavelik** [reply]
Okay understood, so it basically means to run the pipeline more often (we run the pipeline once a week). 

> We (Mend.io) plan to offer a free hosted version of Renovate for Bitbucket Cloud in early 2024, which will run similarly to the current app for github.com, including listening to webhooks and running based on events.

Great looking forward for it.
